### DISCUSSION TITLE
Dolby Vision playback possible with MPV (without HDR profile fallback)?

### DISCUSSION THREAD

**Discussion by definatelynotlhero** [mpv-player/mpv#14776] [General]
I've got a Dolby Vision compatible PC monitor. Does MPV have a way to playback DoVI directly? Like for example when I playback in Windows Media Player I get the Dolby Vision logo and the monitor Dolby Vision settings work. Once I use MPV the monitor fallbacks to the HDR profiles.

**Comment by definatelynotlhero** [comment]
Nop, both Netflix app, and Media Player display Dolby Vision logo on the video itself (not just the logo when you enable dolby vision device profile) and playback Dolby Vision as it should be. 


![image](https://github.com/user-attachments/assets/d360105e-6bb8-416d-99bd-16f4bc907301)

**Reply by kasper93** [reply]
It is possible to switch display to Dolby Vision mode using GPU vendor provided API. Both NVIDIA and AMD provides that. Though, it is likely overkill for mpv. Also the fact that you see DV logo, doesn't mean much about the actual quality and content being passed on.

Example of AMD API https://gpuopen-librariesandsdks.github.io/ags/group__general.html#ga3cdae9f8df77d94be6d9a62ddb066a73 

```
Dolby Vision requires a 8888 UNORM swapchain.
```
I'm also not sure, what format the data should be forwarded in this case and there is not really a way to pass dynamic metadata to the display. So it looks like Dolby Vision enabled mode, but works without much of the dynamic part, although it might be that data should be preprocessed somehow.

I don't know, the information is scarce on this topic and I'm not willing to really go out of my way to implement some fragile vendor specific format signaling. I'm not payed enough for such changes.

I've checked UE code, they enable only `Mode_HDR10_scRGB` mode.

And marketing aside, there is not much to be gained from Dolby Vision output compared to the properly configured HDR10 output for your display, which is already possible in mpv.

**Reply by Frieren924** [reply]
Your understanding is partly right. The fact is that, a dv file normally contains bl (base layer) and rpu data. A bl basically means standard HDR video stream ( not excatlly but basically the same). The problem with today's mpv is that it can't render the video according to rpu file. So if you are ok with watching bl then they are the same ( in some way at least). But if you insist with watching so called "true dolby vision", then rpu render is needed.

**Reply by Frieren924** [reply]
As for the dynamic metadata, dv has a so called "lldv" mode ( low latency dolby vision) , it computes all the metadata in "computing device" and output a simple HDR signal to the playback device.

**Reply by Frieren924** [reply]
I don't know any of how the code works, but I hope my response may have any help.

**Reply by ValeZAA** [reply]
>problem with today's mpv is that it can't render the video according to rpu file

It can. It literally has to?? Both HLG profile 8 --> PQ reshaping and chroma reshaping and standard garbage luma reshaping are supported. What is not supported is NLQ, but that would require scRGB to output 12 bit PQ RGB anyway. Though even 10 bit IPTPQc2 already is already supposedly 11.5 bit of YCbCr, which is around 11.9 bit RGB... That NLQ render is supported in DoviBaker, btw.

>As for the dynamic metadata, dv has a so called "lldv" mode ( low latency dolby vision) , it computes all the metadata in "computing device" and output a simple HDR signal to the playback device.

I think Windows Nvidia driver uses Standard DV.

Also dynamic-source does some of L1 metadata.

**Reply by Frieren924** [reply]
After yesterday's deep dive in to the project, I realized that mpv do support render video stream according to rpu, it's just that you need to add some command in the  config file. So... My bad, you are right haha.

**Reply by Frieren924** [reply]
> > problem with today's mpv is that it can't render the video according to rpu file
> 
> It can. It literally has to?? Both HLG profile 8 --> PQ reshaping and chroma reshaping and standard garbage luma reshaping are supported. What is not supported is NLQ, but that would require scRGB to output 12 bit PQ RGB anyway. Though even 10 bit IPTPQc2 already is already supposedly 11.5 bit of YCbCr, which is around 11.9 bit RGB... That NLQ render is supported in DoviBaker, btw.
> 
> > As for the dynamic metadata, dv has a so called "lldv" mode ( low latency dolby vision) , it computes all the metadata in "computing device" and output a simple HDR signal to the playback device.
> 
> I think Windows Nvidia driver uses Standard DV.
> 
> Also dynamic-source does some of L1 metadata.

As for the stdv part, I think the more precise description is that nv ( or OS) supports both modes . The reason why dolby created such mode is because dolby at that time, is limited by the bandwidth of hdmi port. So they had to use 8bit 444 mode to transport data and the move requires the device to compress the dv data first and then decompress the data in the playback device. From what I know, there are some devices around 2020 uses stdv to support dv playback ( due to the limitation of hdmi2.0 bandwidth), but in recent years, with the pulication of hdm2.0 (2.1), there is no reason to not use lldv now. But dolby is quite a close company. I just can't find any whitepaper or datasheet from dolby offical that can verify the information. If you have any, please let me know, I'm happy to correct my understanding.

**Comment by definatelynotlhero** [comment]
There is definitely a difference in color and quality when using the DoVI files where logo appears on the content in either Netflix\Media player compared to the HDR version in MPV or swapping my monitor to the HDR profiles. My desktop monitor brings the same 3 profiles that usually appear on the TV (DV Dark, Bright and Gaming). Also I am quite sure the DoVI works, because I use it all the time on my Sony A95L TV, and when comparing the DV Dark for example in the content on Netflix, it looks like the right format that I am used to seeing on the TV. 

But I understand your concerns @kasper93. I hope to see in future in MPV support for proper DV playback, nowadays most of the Netflix content (and other platforms) is in the DoVI format, and on my living room TV, I am pretty much always watching DV dark. They look the best to me. However it is true, that Dolby is very closed, and they are annoying when it comes to fixing things.

**Reply by en6ads** [reply]
How do in get MPV to invoke Dolby Access (Dolby Vision Extensions) to decode the video?
Also, How can I pass Dolby Digital Plus Atmos to Dolby Access to decode and render the Atmos?
Thank you.
